vignettes/stata.Rmd

```diff
@@ -0,0 +1,190 @@
+
+```{r}
+mod <- lm(mpg ~ as.factor(cyl) * hp + wt, data = mtcars)
+
+nd <- datagrid(model = mod,
+               cyl = as.factor(c(4, 6, 8)),
+               grid.type = "counterfactual")
+
+mfx <- marginaleffects(mod,
+                       variables = "hp",
+                       newdata = nd) 
+
+aggregate(mfx$dydx, by = list(mfx$cyl), FUN = mean)
+```
+
+<!-- Taken from https://github.com/vincentarelbundock/marginaleffects/issues/226 -->
+
+To obtain standard errors, the “average Jacobian” has to be calculated. The Jacobian is stored in an attribute of the `marginaleffects` object. You can proceed as follows:
+
+```{r}
+J <- attr(mfx, "J")
+J_mean <- aggregate(J, by = list(mfx$cyl), FUN = mean)
+J_mean <- as.matrix(J_mean[, 2:ncol(J_mean)])
+sqrt(diag(J_mean %*% vcov(mod) %*% t(J_mean)))
+```
+
+
+## Average Counterfactual Adjusted Predictions
+
+### Stata
+
+Just like Stata's `margins` command computes average counterfactual marginal effects, it can also estimate _average counterfactual adjusted predictions_.  
+
+Here is an example:
+
+```
+quietly reg mpg i.cyl##c.hp wt
+margins, at(cyl = (4 6 8))
+
+Predictive margins                              Number of obs     =         32
+Model VCE    : OLS
+
+Expression   : Linear prediction, predict()
+
+1._at        : cyl             =           4
+
+2._at        : cyl             =           6
+
+3._at        : cyl             =           8
+
+------------------------------------------------------------------------------
+             |            Delta-method
+             |     Margin   Std. Err.      t    P>|t|     [95% Conf. Interval]
+-------------+----------------------------------------------------------------
+         _at |
+          1  |   17.44233   2.372914     7.35   0.000     12.55522    22.32944
+          2  |    18.9149   1.291483    14.65   0.000     16.25505    21.57476
+          3  |   18.33318   1.123874    16.31   0.000     16.01852    20.64785
+------------------------------------------------------------------------------
+```
+
+Again, this is what Stata does in the background:
+
+  1. It duplicates the whole dataset 3 times and sets the values of `cyl` to the three specified values in each of those subsets.
+  2. It calculates predictions for that large grid.
+  3. It takes the average prediction for each value of `cyl`.
+  
+In other words, average counterfactual adjusted predictions as implemented by Stata are a hybrid between predictions at the observed values (the default in `marginaleffects::predictions`) and predictions at representative values. 
+
+### marginaleffects
+
+You can estimate average counterfactual adjusted predictions with `predictions()` by, first, setting the `grid.type` argument of `data.grid()` to `"counterfactual"` and, second, by averaging the predictions, for example using `aggregate()` or `dplyr::summarise()`.
+
+```{r}
+mod <- lm(mpg ~ as.factor(cyl) * hp + wt, data = mtcars)
+
+nd <- datagrid(model = mod,
```

Comment:
If you use `datagrid()` inside the function call directly with the `newdata` argument, you won't need to specify the `model` argument manually. Also, if you specify inside function call, you probably don't need the `as.factor()`, since `marginaleffects` will coerce it to factor if it sees that it is a factor in the original data.